### ISSUE TITLE
This error is always displayed about me running this project：

### ISSUE THREAD

**Issue by ZYYYY123324** [siposcsaba89/ocam-calib-cpp#2]
错误	LNK2019	无法解析的外部符号 "double __cdecl calib_converter::convertOcam2Mei(class std::vector<double,class std::allocator<double> > const &,class std::vector<double,class std::allocator<double> > const &,class Eigen::Matrix<double,2,1,0,2,1> const &,class Eigen::Matrix<double,2,1,0,2,1> const &,class Eigen::Matrix<float,3,3,0,3,3> &,class std::array<float,5> &,float &)" (?convertOcam2Mei@calib_converter@@YANAEBV?$vector@NV?$allocator@N@std@@@std@@0AEBV?$Matrix@N$01$00$0A@$01$00@Eigen@@1AEAV?$Matrix@M$02$02$0A@$02$02@5@AEAV?$array@M$04@3@AEAM@Z)，函数 "bool __cdecl runCalibration(class std::vector<class std::vector<class cv::Point_<float>,class std::allocator<class cv::Point_<float> > >,class std::allocator<class std::vector<class cv::Point_<float>,class std::allocator<class cv::Point_<float> > > > >,class cv::Size_<int>,class cv::Size_<int>,enum Pattern,float,float,int,class cv::Mat &,class cv::Mat &,class std::vector<class cv::Mat,class std::allocator<class cv::Mat> > &,class std::vector<class cv::Mat,class std::allocator<class cv::Mat> > &,class std::vector<float,class std::allocator<float> > &,double &)" (?runCalibration@@YA_NV?$vector@V?$vector@V?$Point_@M@cv@@V?$allocator@V?$Point_@M@cv@@@std@@@std@@V?$allocator@V?$vector@V?$Point_@M@cv@@V?$allocator@V?$Point_@M@cv@@@std@@@std@@@2@@std@@V?$Size_@H@cv@@1W4Pattern@@MMHAEAVMat@4@3AEAV?$vector@VMat@cv@@V?$allocator@VMat@cv@@@std@@@2@4AEAV?$vector@MV?$allocator@M@std@@@2@AEAN@Z) 中引用了该符号	ocam	D:\fisheye_cali\fisheye-calib.obj	1

**Comment by siposcsaba89** [comment]
@ZYYYY123324 

It seems like you have some linker error, maybe you don't use cmake, so  these are all the files you need to be able to compile and link the program:

- ocam-fisheye-calib
- fisheye-calib.cpp
- ocam_calibration.h
- ocam_calibration.cpp
- calib_converter/calib_converter.h
- calib_converter/calib_converter.cpp

Maybe you missing some of them?

**Comment by ZYYYY123324** [comment]
OK.Let me try again!Thanks

**Comment by ZYYYY123324** [comment]
Hello, sorry to bother you again. I am a student who has just come into contact with this kind of project. I want to ask you new questions.1）Is the whole project based on Linux or windows? 2）“ocam-fisheye-calib”Is it the lines of Linux instructions in readme（his is a camera calibration sample.
Usage: calibration
     -w=<board_width>         # the number of inner corners per one of board dimension
     -h=<board_height>        # the number of inner corners per another board dimension
     ...
     [input_data]             # input data, one of the following:
                              #  - text file with a list of the images of the board
                              #    the text file can be generated with imagelist_creator
                              #  - name of video file with a video of the board
                              # if input_data not specified, a live view from the camera is used）

**Comment by siposcsaba89** [comment]
Hi @ZYYYY123324!

1. It is a cross platform application, so it works with Linux and Windows as well
2. Yes those lines are the same for windows and Linux ocam-fisheye-calib binaries

**Comment by ZYYYY123324** [comment]
Thank you again for your help. I'll try again！ @siposcsaba89

**Comment by somdats** [comment]
@siposcsaba89 I have calibrated a fish-eye lens camera with scaramuzza toolbox and now using your functionalities to convert it into MeiModel so that I can use it in OpenCV compatible form. Howe
![undistorted_perspective_test_conv](https://user-images.githubusercontent.com/4468850/160354772-242d0769-15d1-40c0-a9b5-e8fadde1d3b3.jpg)
it is the result with converter
![undistorted_perspecti
![25032022_new0](https://user-images.githubusercontent.com/4468850/160354809-77b81715-1513-4d56-81d0-8cc56d9e943a.jpg)
it is the actual image
ve_test_25032022_sc](https://user-images.githubusercontent.com/4468850/160354788-2865444e-2473-4598-b269-410ca7107b06.jpg)
it is the result with scaramuzza toolbox undistortion

ver, the results are vastly different If I undistort an image using the actual Scaramuzza tool compared with the undistorted image after using the converter.  I made some modifications to your converter, for instance, Board size and square size so that it matches my pattern.
Any lead would be helpful, I did not understand what exactly num_table _count does
Here is the calib result with scaramuzza
#polynomial coefficients for the DIRECT mapping function (ocam_model.ss in MATLAB). These are used by cam2world

5 -4.754486e+02 0.000000e+00 7.054599e-04 6.083052e-08 1.427329e-10 

#polynomial coefficients for the inverse mapping function (ocam_model.invpol in MATLAB). These are used by world2cam

9 755.539213 533.425006 108.907522 92.422761 46.680117 24.788708 15.143199 5.684140 0.846350 

#center: "row" and "column", starting from 0 (C convention)

577.703946 983.424044

#affine parameters "c", "d", "e"

1.000194 -0.058458 0.058290

#image size: "height" and "width"

1096 1948

Here is the output from your converter tool
Num tables total: 58
sum trans error in meter: 4.77641, 
avg_err: 0.0823519
sum rad error in degree: 1906.89, 
avg_err: 32.8773Converted parameters:
K:
[1181.330885702317, 0, 983.4547450854144;
 0, 1181.014283407671, 577.8375036140573;
 0, 0, 1]
xi:
[1.479367708605731]
D:
[-0.2698151761484431, 0.3409557309874347, -2.481265261341467e-05, 2.997549119019755e-05]
rms:0.00109052,
 num_total_pts: 4283
